test/Api.Tests/ControllerTests/UserControllerTest.cs

```diff
@@ -0,0 +1,135 @@
+                .Ok()
+                .WithStatusCode(200)
+                .WithModelOfType<UserResult>()
+                .Passing(m => m.Email == _tester.Admin.Email);                
+        }
+
+        /// <summary>
+        /// Should return 200 when trying to get user details
+        /// </summary>
+        [Fact]
+        public void GetUserTest200ForStudent()
+        {
+            _tester.TestController()
+                .Calling(c => c.Get(_tester.Student.Email))
+                .ShouldReturn()
+                .Ok()
+                .WithStatusCode(200)
+                .WithModelOfType<UserResult>()
+                .Passing(m => m.Email == _tester.Student.Email);
+        }
+
+        /// <summary>
+        /// Should return 404 when trying to get user details
+        /// </summary>
+        [Fact]
+        public void GetUserTest404()
+        {
+            _tester.TestController()
+                .Calling(c => c.Get("hello@hipapp.de"))
+                .ShouldReturn()
+                .NotFound();
+        }
+
+        #endregion
+
+        #region PUT user tests
+
+        /// <summary>
+        /// Should return code 200 if user is updated
+        /// </summary>
+        [Fact]
+        public void PutUserTest200()
+        {
+            _tester.TestController()
+                .Calling(c => c.Put(_tester.Admin.Email, UserFormModel))
+                .ShouldHave()
+                .DbContext(db => db.WithSet<User>(user =>
+                    user.Find(_tester.Admin.Id).FirstName.Equals(UserFormModel.FirstName) &&
+                    user.Find(_tester.Admin.Id).FirstName.Equals(UserFormModel.FirstName)
+                ))
+                .AndAlso()
+                .ShouldReturn()
+                .Ok()
+                .WithStatusCode(200);
+        }
+
+        /// <summary>
+        /// Should return bad request when trying to update user with invalid role
+        /// </summary>
+        [Fact]
+        public void PutUserTest400()
+        {
+            UserFormModel.Role = "InvalidRole";
+
+            _tester.TestController()
+                .Calling(c => c.Put(_tester.Admin.Email, UserFormModel))
+                .ShouldReturn()
+                .BadRequest()
+                .WithStatusCode(400);
+        }
+
+        /// <summary>
+        /// Should return 403 when student trying to update user
+        /// </summary>
+        [Fact]
+        public void PutUserTest403()
+        {
+            _tester.TestController(_tester.Student.Email)
+                .Calling(c => c.Put(_tester.Admin.Email, UserFormModel))
+                .ShouldReturn()
+                .StatusCode(403);
+        }
+
+        /// <summary>
+        /// Should return user not found when a new user is trying to get updated
+        /// </summary>
+        [Fact]
+        public void PutUserTest404()
+        {
+            _tester.TestController()
+                .Calling(c => c.Put("user@hipapp.de", UserFormModel))
```

Comment:
write `"nonexistinguser@hipapp.de"` or sth. like that, so it's clear that the user does not exist